openscm/definitions/emissions_units.txt

```diff
@@ -0,0 +1,58 @@
+yr = 1 * year
+kt = 1000 * t
+
+C = [carbon_emissions]
```

Comment:
Shouldn't we define the actual weight as units, e.g. "tC" and "gC"? That way we can avoid the otherwise necessary space between "t" and "C" and are actually defining units?
I would strongly suggest that.

Reply:
I thought about this too. However I think that blends two things which are actually separate (whatever the outcome, I agree that this conversation needs to go in the docs somewhere).

For example, when you buy bananas, you buy a kilo of bananas. The units are kg bananas, not kgbananas. I think an emissions unit is three separate components, the mass, what the stuff is and the time period (/yr, /day etc.). 

Blending the first two means that we have to define each of the masses for every new gas e.g. "tC, MtC, GtC, PgC...", "tCO2, MtCO2, GtCO2, PgCO2" and the conversions between all of these things. This will be a real pain and hard to keep track of as the number of combinations is much bigger because you have to two things to keep track of (mass and the stuff) rather than keeping them separate and letting Pint do the combinations for us.

I also think blending the two makes it less clear what these emissions units actually are. When you think about it, they're super weird. They're not simply atomic masses, if they were one could convert between C and N2O with no issue. They're really a counting measure which is particular to emissions. Hence we have this funny (or at least it seems funny to me now) convention where C and CO2 are interchangeable (despite being fundamentally different species), but C and N2O are not. In any case, if we keep the mass separate from the emissions, it's easy for us to say that you can swap C and CO2 but not e.g. C and N2O (and hence CO2 and N2O by consequence thanks to Pint's handling). It then also makes it much easier to work with contexts as we only need to define gas to gas conversions and don't need to define the conversions between all the masses of all the different gases for each new context.

Reply:
The prefixes (M, G, P, ...) should be handled be the unit library (Pint also does so, I hope?!), so I don't see a problem there. Can we otherwise circumvent forcing people to put a space between weight unit and gas "unit"?

Reply:
> Can we otherwise circumvent forcing people to put a space between weight unit and gas "unit"?

Sure. We just wrap whatever unit conversions we want to do with something like 

```python
def convert_unit(in_unit, out_unit):
    if in_unit.startswith("t", "Mt", "Gt", "g", "Pg", "Mg", "Gg"):  # etc.
        in_unit = "{} {}".format(in_unit[:2], in_unit[2:])
    # the rest
```

Having said that, I still think this sends the wrong signal. They're not one unit, they're two.

Reply:
Hmm, you're right, then let's follow the philosophy of the gas type as a separate unit... (`convert_unit` would only take care of some weights and the very beginning of the full unit though...)

Reply:
yep `convert_unit` would need some serious testing for us to be sure it covers all the uses cases, I'm happy for that to go in the bin

Reply:
Idea for a compromise: We could define `gC = g * C`, `tC = t * C` et al (so two additional lines per new gas unit)...

Reply:
genius! We could also define those units dynamically... Let me have a fiddle today/tomorrow

Reply:
Awesome, that looks great! And feel more natural and like what people are already using.

Reply:
Other question here then, do we want to still use the default registry or should we restrict only to units we define to avoid unexpected collisions (like the weird thing with h being planck constant by default?)

Reply:
Hmm, is there an SI registry? It's nice to have all the standard units in there already... Otherwise, let's just overwrite the default where needed.
We should have a list of units we want to be supported by the OpenSCM standard (e.g., for other implementations) in the docs/the standard.

Reply:
+1 for sensible defaults and no weird units (if that means we only use ours by default that's fine I'd say)

Reply:
> Hmm, is there an SI registry?

I don't think so, we could split it out in Pint (again not hard, just more fiddling)

> We should have a list of units we want to be supported by the OpenSCM standard (e.g., for other implementations) in the docs/the standard.

Good idea

> Otherwise, let's just overwrite the default where needed.

Cool

Reply:
@znicholls fyi, I have some time right now and will quickly include the additional units...